odoo/modules/registry.py

```diff
@@ -282,6 +284,8 @@ def setup_models(self, cr):
         for model in models:
             model._setup_complete()
 
+        del self._m2m
```

Comment:
I think we can define `self._m2m` right before `_setup_fields` and delete it right after, no?